### DISCUSSION TITLE
Property list (.plist) export / sync with macOS text replacement

### DISCUSSION THREAD

**Discussion by krystof-k** [espanso/espanso#1314] [Feature requests and Ideas]
It would be great if it could be possible to convert the YAML to macOS property list (.plist) file so that it can be easily imported into the text replacement in system preferences ([see help](https://support.apple.com/guide/mac-help/back-up-and-share-text-replacements-on-mac-mchl2a7bd795/mac)), for example using `espanso export -f plist` command.

E.g.:

```yaml
matches:
  - trigger: ":espanso"
    replace: "Hi there!"
```

to

```yml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<array>
  <dict>
    <key>phrase</key>
    <string>Hi there!</string>
    <key>shortcut</key>
    <string>:espanso</string>
  </dict>
</array>
</plist>
```

Automatic sync would be even more awesome, but it would be probably quite challenging to develop due to the need hacking into Apple's private frameworks (see unfortunately no longer maintained [rodionovd/shortcuts](https://github.com/rodionovd/shortcuts) repo).

**Comment by federico-terzi** [comment]
Hey @krystof-k,

Thank you for the feedback! Indeed, this could be very useful. As you suggested, I'm only worried that working with Apple's private APIs would be a maintenance challenge (though it would be great as shortcuts could be synced with iPhones)
If there's enough demand from the community we'll definitely investigate

Cheers :)

**Comment by smeech** [comment]
It's been very quiet on this topic for a long time, which is a shame as it's a fairly simple scripting task, even if it's not something that would be built into Espanso.
